### ISSUE TITLE
Add sankey style confusion matrix visualization

### ISSUE THREAD

**Issue by Nadav-Zilberberg** [Nadav-Zilberberg/scikit-learn-test#25]
Describe the workflow you want to enable
Confusion matrices can be displayed as a colored matrix using the [ConfusionMatrixDisplay](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ConfusionMatrixDisplay.html#sklearn.metrics.ConfusionMatrixDisplay) class.

confusion_matrix

However color scaling is hard to interpret, and spatial cues are easier to help interpret quantities than color variations.
The number represented in each box represent absolute ones, while one may be interested in either row based, column based, or complete matrix sum based normalization.
Plus it gets even harder to read for multiclass classification.

Describe your proposed solution
I would propose introducing sankey like plots to visualize the confusion matrix data, using a ConfusionMatrixSankeyDisplay class.
The qualitative information is displayed by different colors (easy to interpret), while actual amounts in each cell are represented by the size of the flows (easier to interpret quantitatively than colorscale variations).
On the left size one can see the number of occurrence of each label in the ground truth data (confusion matrix row marginals). On the right side the number of occurrence of each label in the predictions (confusion matrix column marginals). Each flow represents both row-normalized (left side) and column-normalized (right side) at the same time, and could be labeled with the actual absolute number of examples in each confusion matrix cell.
Interpretation is straightforward even in the multiclass case.

There exist a matplotlib based implementation doing almost exactly this in the small [pySankeyBeta](https://github.com/Pierre-Sassoulas/pySankey) package. Here is a sample from its readme:
sankey

Describe alternatives you've considered, if relevant
No response

Additional context
No response
